ocfweb/docs/docs/staff/backend/kubernetes.md

```diff
@@ -0,0 +1,190 @@
+Although, reading this document will help you both deploy services in the OCF
+Kubernetes cluster and debug issues when they arise.
+
+## Kubernetes
+
+Kubernetes is a container orchestration system open sourced by Google. Its main
+purpose is to schedule services to run on a cluster of computers while
+abstracting away the existence of the cluster from the services. The design of
+Kubernetes is loosely based on Google's internal orchestration system Borg.
+Kubernetes is now maintained by the [Cloud Native Computing Foundation][cncf],
+which is a part of the Linux Foundation. Kubernetes can flexibly handle
+replication, impose resource limits, and recover quickly from failures.
+
+## Kubernetes Cluster
+
+A Kubernetes cluster consists of "master" nodes and "worker" nodes. In short,
+master nodes share state to manage the cluster and schedule jobs to run on
+workers. It is considered best practice to run an odd number of masters, and
+currently our cluster has three masters.
+
+### Masters
+
+Kubernetes masters share state via a distributed key-value store. The reason
+for the key-value store is high availability: if one master goes down, a new
+master node will subsume control of the cluster in a leader election to ensure
+that no services incur downtime. [etcd][etcd-io] is an
+implementation of the [Raft][raft] protocol chosen by the Kubernetes project
+to act as the key-value store. It's three main goals are:
+
+1. Leader elections in case of failure.
+2. Log replication across all masters.
+3. Ensuring log integrity across all masters.
+
+### Why the odd number of masters?
+
+Consider a cluster of *N* members. When masters form quorum to agree on cluster
+state, quorum must have _at least_ ⌊*N*/2⌋+1 members. Every new odd number in a
+cluster with *M* > 1 masters adds one more node of fault tolerance.  Therefore,
+adding an extra node to an odd numbered cluster gives us nothing. If interested
+read more [here][failure-tolerance].
+
+### Workers
+
+Workers are the brawn in the Kubernetes cluster. While master nodes are
+constantly sharing data, managing the control plane, and scheduling services,
+workers primarily run [pods][pod]. `kublet` is the service that executes pods
+as dictated by the control plane, performs health checks, and recovers from pod
+failures should they occur. Workers also run an instance of `kube-proxy`, which
+forwards control plane traffic to the correct `kublet`.
+
+### Pods
+
+In the Kubernetes world, pods are the smallest computing unit. A pod is made up
+of one or more containers. The difference between a pod and a standalone
+container is best illustrated by an example. Consider [ocfweb][ocfweb]; it is
+comprised of several containers—the web container, static container, and worker
+container.  In Kubernetes, together these three containers form one pod, and it
+is pods that can be scaled up or down. A failure in any of these containers
```

Comment:
These don't necessarily have to be in the same pod, and when we actually deploy ocfweb I'd consider putting static files in a separate pod.

I guess you can keep this here for now though, if that ends up happening I'll change it.